src/bflib_mouse.cpp

```diff
@@ -161,46 +161,6 @@ TbResult LbMouseOnMove(struct TbPoint shift)
   return Lb_SUCCESS;
 }
 
-/** Converts mouse coordinates into relative shift coordinates.
- *
- * @param pos Pointer to the structure with source point, and where result is placed.
- */
-void MouseToScreen(struct TbPoint *pos)
-{
-  // Static variables for storing last mouse coordinated; needed
```

Comment:
This function not only clips mouse but also scales it to virtual 640x480

Reply:
> This function not only clips mouse but also scales it to virtual 640x480

Actually not, when I played in 1366x768 `GetMouseWindow` returned (0, 0, 1365, 767), so this line:
```c
pos->x = ((pos->x - mx) * (long)lbDisplay.MouseMoveRatio)/256;
```
worked like this: let's say x = 500, 500 - 0 = 500, 500 * MouseMoveRatio/256 = 500, because MouseMoveRation hardcoded to 256 inside `LbScreenInitialize`
```c
LbMouseChangeMoveRatio(256, 256);
```
So, this part basically does nothing, and as a result `pos` coordinates are turned into delta from previous movement, which are exactly the same as I get from `ev->motion.xrel` and `ev->motion.yrel` in `process_event`:
```c
    case SDL_MOUSEMOTION:
        mousePos.x = ev->motion.xrel;
        mousePos.y = ev->motion.yrel;
        mouseControl(MActn_MOUSEMOVE, &mousePos);
        break;
```
And clipping also happens inside `MouseStateHandler::SetPosition` too:
```c
...
    mx = x;
    if (x < lbDisplay.MouseWindowX)
    {
      mx = lbDisplay.MouseWindowX;
    } else
    if (x >= lbDisplay.MouseWindowX+lbDisplay.MouseWindowWidth)
    {
      mx = lbDisplay.MouseWindowWidth + lbDisplay.MouseWindowX - 1;
    }
...
```
I can put this func back, but it'll not change anything.
Actually `LbMouseOnMove` is not needed too and we can set new coords inside `mouseControl` like this:
```c
pointerHandler.SetMousePosition(pos->x, pos->y)
```
In this case SDL will keep track and bounds of mouse position itself. However removing `LbMouseOnMove` caused some glitches with creature view (first person) which I was unable to resolve, thus leaving things as is for now.
But I'm seriously tempted to clean legacy code inside `bflib_mouse.c` and `bflib_video.c` and `vidmode.c`.
All these `setup_screen_mode_zero`, `reenter_video_mode` etc are completely unnecessary. However for now I decided to stay minimal.

Reply:
@SimLV do you agree with this analysis, or do you still see problems with merging like this?

Reply:
Ok

Reply:
"but also scales it to virtual 640x480" - this seems to be the case.

lbDisplay.MouseMoveRatio is changed based on the display resolution (this can be observed from the log file). So the following 2 lines are required **at the least**, to restore the mouse sensitivity to what it was before this pull request. 
```
pos->x = (pos->x * (long)lbDisplay.MouseMoveRatio)/256;
pos->y = (pos->y * (long)lbDisplay.MouseMoveRatio)/256;
```
I put back the MouseToScreen function, with just this functionality, and previous sensitivity is restored. But this doesn't work well with altinput mode. The SDL_WarpMouseInWindow probably corrected this before. I'll keep looking, but if you have any input Gecko, that would be great.

Reply:
`SDL_WarpMouseInWindow` in SDL2 generates mouse movement event which is not good for us. `SDL_WarpMouse` was used to `center` mouse pointer so that we could receive continuous movement data. Currently we get relative movement data with `ev->motion.xrel`. Now I see that `base_mouse_sensitivity` that's affected by `POINTER_SENSITIVITY` config parameter actually affects the speed with which cursor moved, funny that it's like `1` for 1366x768. I propose multiplying `ev->motion.xrel` by `base_mouse_sensitivity` to receive desired effect, but yes this will cause problems with altinput. I'll look into this, but alas currently I cannot test in 1920x1080.